### ISSUE TITLE
Import Validator indicates import validation error if successfully imported module has import validation error

### ISSUE THREAD

**Issue by jongiddy** [DamnWidget/anaconda#254]
If module A imports a nonexistent module B, the Import Validator correctly indicates that it
can't import module B.

However, if module C then imports module A, which is in the path, the Import Validator incorrectly indicates that it can't import module A.

e.g. `good.py` (module A)

```
# Uncomment this import to see error.
# import nonexist


def foo():
    pass
```

`test.py` (module C)

```
from good import foo

foo()
```

Assuming `good.py` is in the Python path, then `test.py` should have no import validation errors.  However, it does if the broken import in `good.py` is uncommented.

**Comment by DamnWidget** [comment]
Just one question.
How a module that has an import error can be successfully imported by other module?

I mean, if we have three modules as in your example, A, B and C, and A has an import error, and B imports A, C will never import B successfully as B throws an `ImportError` exception so it fails to compile to bytecode so import B will never be a success operation as A has to be compiled by the interpreter before B is even being processed.

I think the behavior is just the expected one.

**Comment by jongiddy** [comment]
I do see your point. However, during development, I would like two cases to be distinguished:

1. The imported module is not present in the module search path (the search path is misconfigured, a dependency is not installed, or the package name is mispelt).

2. The imported module is present in the search path, but cannot be imported due to its own errors (probably this is my code as well, and I'd rather this be indicated in the source of the imported module).

I'm not sure how to actually distinguish the two in the Anaconda code. I had a look at `jedi/evaluate/imports.py`, which says "there's not any actual importing done", but I have yet to figure out whether that allows the dual behaviour described.

Given this description of the behaviour, maybe this could be considered a feature request rather than a bug?

**Comment by DamnWidget** [comment]
Yeah, probably it would be considered a feature request, anyway I am not sure how valuable it is or even if it's possible to do it at all.

**Comment by DamnWidget** [comment]
Fixed in release v2.1.10
